tests/components/cover/test_init.py

```diff
@@ -1,5 +1,123 @@
 """The tests for Cover."""
 import homeassistant.components.cover as cover
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    CONF_PLATFORM,
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_TOGGLE,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.setup import async_setup_component
+
+
+async def test_services(hass, enable_custom_integrations):
+    """Test the provided services."""
+    platform = getattr(hass.components, "test.cover")
+
+    platform.init()
+    assert await async_setup_component(
+        hass, cover.DOMAIN, {cover.DOMAIN: {CONF_PLATFORM: "test"}}
+    )
+    await hass.async_block_till_done()
+
+    ent1, ent2, ent3, ent4 = platform.ENTITIES
+
+    # Test init
+    assert is_open(hass, ent1.entity_id)
+    assert is_open(hass, ent2.entity_id)
+    assert is_open(hass, ent3.entity_id)
+    assert is_open(hass, ent4.entity_id)
```

Comment:
Add a comment explaining the difference between the 4 covers.
Do we test the toggle behavior of a cover which doesn't support STOP?

Reply:
I'd suggest to modify `MockCover` to not calculate its `supported_features` but to instead accept it as a parameter to the constructor, like `MockLight` does.

With that changed, we can easily test the behavior of covers without STOP support.

Reply:
Good Idea! MockCover now supports feature setup on creation.

But the Concept Problem still exists in CoverEntity on default behaviour.